Automattic/wp-calypso
# Update Cookie Banner

#### Proposed Changes

- Add @automattic/privacy-toolset to wp-calypso
- Remove GdprBanner and replace it with CookieBanner (SSR and async)
- Include checking both GDPR and CCPA zones
- Support new cookie structure server-side

#### Testing Instructions

> Disclaimer: this PR changes env flag from `gdpr-banner` to `cookie-banner`. Please consider that in the testing plan.

* Checkout changes
* Modify the config for your test environment to ensure that cookie-banner is set to true

##### Testing client-side Cookie Banner
* Run the calypso environment (`yarn start`)
* Go to any authenticated calypso page e.g., `http://calypso.localhost:3000/me/privacy?flags=cookie-banner`
* Change your cookies to CCPA region `country_code=US; region=California`
* Ensure you don't have `sensitive_pixel_option` and `sensitive_pixel_options` cookies
* Reload the page
* You should see the cookie banner floating at the bottom of the viewport

<img width="1321" alt="CleanShot 2022-11-30 at 19 17 03@2x" src="https://user-images.githubusercontent.com/8419292/204942520-dc2aee91-efb4-4e91-9794-bbfa41853598.png">

* Click "Customize"
* You should see the cookie banner with bucket options

<img width="969" alt="CleanShot 2022-12-02 at 08 26 28@2x" src="https://user-images.githubusercontent.com/8419292/205314964-1f6dd662-c6e9-4cc1-b415-d9963255f523.png">

* Click "Accept selection"
* Reload the page
* Cookie Banner should not load again

##### Testing server-side Cookie Banner

* (if you don't have VPN)
* Run the calypso environment (`yarn start`)
* Load `http://calypso.localhost:3000/log-in` as a logged-out user
* Change your cookies to GDPR region `country_code=NL` (CCPA is not yet supported on the server)
* (if you have VPN)
* Set VPN to any of the EU countries, e.g., France.
* (continue)
* Run calypso environment with server-side `yarn build-server && yarn start`)
* Ensure you don't have `sensitive_pixel_option` and `sensitive_pixel_options` cookies
* Load `http://calypso.localhost:3000/log-in` as a logged-out user
* Ensure the cookie bar gets displayed as part of the first render
* Click the Accept button
* Ensure the cookie bar animates away
* Reload the page
* Ensure the cookie bar does not get displayed as part of the first render

#### Pre-merge Checklist
- [x] Ensure the changes do not impact SSR negatively
- [x] Merge and publish @automattic/privacy-toolset@1.0.1 (https://github.com/Automattic/wp-calypso/pull/70709)

#### After-merge Checklist
- [x] Refactor function `refreshCountryCodeCookieGdpr` -> `refreshGeoCookies`
- [ ] Update wp-calypso flag docs from `gdpr-banner` -> `cookie-banner`

Related to #69304 (this PR replaces the previous one)
